### PR TITLE
Feature/histogramupdate

### PR DESCRIPTION
**Pull request by petersteneteg**
Histogram calculation fixes

And some minor refactorings

### CODE REVIEW

File: include/inviwo/core/datastructures/histogram.h [hunk 1]
```diff
@@ -42,71 +44,116 @@ enum class HistogramMode { Off, All, P99, P95, P90, Log };
...
+    // a double type with the same extent as T
+    using D = typename util::same_extent<T, double>::type;
+    // a size_t type with same extent as T
+    using I = typename util::same_extent<T, size_t>::type;
+
+    constexpr size_t extent = util::rank<T>::value > 0 ? util::extent<T>::value : 1;
+
+    // check whether number of bins exceeds the data range only if it is an integral type
+    if constexpr (!util::is_floating_point<typename util::value_type<T>::type>::value) {
+        bins = std::min(bins, static_cast<std::size_t>(dataRange.y - dataRange.x + 1));
```
Comment (martinfalk) [inline include/inviwo/core/datastructures/histogram.h:109]: good catch!

File: include/inviwo/core/datastructures/histogramtools.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#include <memory>
+#include <vector>
+
+namespace inviwo {
+
+class HistogramSupplier;
+
+class IVW_CORE_API HistogramCalculationState {
+public:
+    void whenDone(std::function<void(const HistogramContainer&)> callback);
```
Comment (martinfalk) [inline include/inviwo/core/datastructures/histogramtools.h]: weird order of functions

File: src/core/datastructures/volume/volume.cpp [hunk 3]
```diff
@@ -28,26 +28,26 @@
...
     , defaultDataFormat_(defaultFormat)
     , defaultSwizzleMask_(defaultSwizzleMask) {}
 
 Volume::Volume(std::shared_ptr<VolumeRepresentation> in)
-    : Data<Volume, VolumeRepresentation>()
-    , StructuredGridEntity<3>()
-    , MetaDataOwner()
+    : Data<Volume, VolumeRepresentation>{}
+    , StructuredGridEntity<3>{}
+    , MetaDataOwner{}
```
Comment (martinfalk) [inline src/core/datastructures/volume/volume.cpp:50]: HistogramSupplier{}

**Review by martinfalk — APPROVED** [review]

**Review by martinfalk — APPROVED** [review]

File: include/inviwo/core/ports/inport.h [hunk 4]
```diff
@@ -100,41 +100,29 @@ class IVW_CORE_API Inport : public Port {
...
 
-    // clang-format off
     /**
      * The on change call back is invoked before Processor::process after a port has been connected,
      * disconnected, or has changed its validation level. Note it is only called if process is also
      * going to be called.
      */
-    template <typename T>
-    [[deprecated("was declared deprecated. Use `onChange(std::function<void()>)` instead")]]
-    const BaseCallBack* onChange(T* o, void (T::*m)());
```
Comment (ResearchDaniel) [inline include/inviwo/core/ports/inport.h:111]: We should bump the version when making this change. This will affect old code

**Pull request merged**